pymonetdb/filetransfer/uploads.py

```diff
@@ -0,0 +1,245 @@
+"""
+Classes related to file transfer requests as used by COPY INTO ON CLIENT.
+"""
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0.  If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright 1997 - July 2008 CWI, August 2008 - 2016 MonetDB B.V.
+
+
+from io import BufferedIOBase, BufferedWriter, RawIOBase, TextIOBase, TextIOWrapper
+from typing import Any, Optional, Union
+from pymonetdb import mapi as mapi_protocol
```

Comment:
why not just do `from pymonetdb.mapi import Connection` and use that as type?

Reply:
It gave circular import problems